### DISCUSSION TITLE
Bug/Juggling Commits #2  False solution

### DISCUSSION THREAD

**Discussion by sencercebeci** [pcottle/learnGitBranching#1042] [Show and tell]
I found a bug i think,

In Juggling Commits #2  question, we had to use cherry pick and amend.
The way should be : 

$ git checkout main
$ git cherry-pick C2
$ git commit --amend
$ git cherry-pick C3

But if you write that, it works too;

$ git checkout main
$ git cherry-pick c2 c3

This is false because the show goal has another picture for solution.
your welcome <3

**Comment by pcottle** [comment]
Those two commands don't seem to solve the level. We are already checked out on main, and then we get this error:

<img width="411" alt="Screen Shot 2022-12-15 at 9 27 40 AM" src="https://user-images.githubusercontent.com/1135007/207914924-2261ab66-9b63-4b5e-a0e0-88468dce2f3c.png">

Regardless, we have a pretty loose way of comparing trees to the desired solution. It doesn't have to look *exactly* like the solution, but rather just have the same properties -- like the same commits, their order, and their relative amend order (the hashes). so its not a bug if you find a solution that doesnt exactly match our solution tree

**Reply by sencercebeci** [reply]
We are not already checked out on main ,we are on caption. 
We need to do "it checkout main" first.

It's just not same as our goal thats why i want to tell it.
